### ISSUE TITLE
2018-Q2 2.1 Ship webcompat system addon v2

### ISSUE THREAD

**Issue by miketaylr** [mozilla/webcompat-team-okrs#8]
* [x] Land the code in Desktop (61 train?) [Bug 1386807](https://bugzilla.mozilla.org/show_bug.cgi?id=1386807)
* [x] Define site patching release process with RelMan
* [ ] Build 6 site patches:
   * [x] bankofamerica.com - [tracking bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1472075), [site patch PR](https://github.com/mozilla/webcompat-addon/pull/24)
   * [x] election.gov.np - [tracking bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1472081), [site patch PR](https://github.com/mozilla/webcompat-addon/pull/25)
   * [x] histography.io - [tracking bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1457335), [site patch PR](https://github.com/mozilla/webcompat-addon/pull/23)
   * [x] ib.absa.co.za - [tracking bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1452707), [site patch PR](https://github.com/mozilla/webcompat-addon/pull/22)
   * [ ] (Android) https://bugzilla.mozilla.org/show_bug.cgi?id=755590 - sites.google.com
   * [ ] (Android) https://github.com/webcompat/web-bugs/issues/13154 - m.imgur.com
* [ ] Ship the site patches above to their targets
* [ ] Get add-on code landed in Fennec ([bug 1338586](https://bugzilla.mozilla.org/show_bug.cgi?id=1338586))

Relevant bug that enabled Fennec to handle system addons: https://bugzilla.mozilla.org/show_bug.cgi?id=1260213

**Comment by miketaylr** [comment]
Please use this comment to track candidate site patch bugs:

1. [ ] https://bugzilla.mozilla.org/show_bug.cgi?id=1442050

**Comment by miketaylr** [comment]
Dennis: the code is finished. Waiting on some small review feedback and new QA contact to do tests. 61 soft freeze is in 9 days (April 26th).

**Comment by denschub** [comment]
Got a second r+ by :mconley, approval from QA arrived, patch is waiting for checking. Bumping the confidence up a fair bit.

**Comment by miketaylr** [comment]
> Define site patching release process with RelMan (but TBD)

https://wiki.mozilla.org/Compatibility/Go_Faster_Addon/Override_Policies_and_Workflows#Checklist_for_shipping_a_site_patch

**Comment by miketaylr** [comment]
Patch has landed, but there's some perf regression concerns and possibly related intermittents to fix.

**Comment by denschub** [comment]
Related:

* https://bugzilla.mozilla.org/show_bug.cgi?id=1455796 - `Intermittent dom/base/test/browser_messagemanager_loadprocessscript.js | Should get back to the base number of processes at this point (3 or 4)`
  * 2018-05-02: Patch to fix the test proposed, waiting for review.
* https://bugzilla.mozilla.org/show_bug.cgi?id=1457538 - `14.37% tp5o responsiveness (linux64) regression`
  * 2018-05-02: That's really hard to fix, actually. Started a discussion to check if we can just ignore it, and asking :aswan for his opinion.

**Comment by denschub** [comment]
I ticked the "land source to desktop" checkbox, as it looks like we'll stick.

The intermittent mentioned earlier were resolved by adjustments in the test last week. Those changes were uplifted into the 61 beta, and failure rate for that test is now back at 0.

The performance issue was closed as a WONTFIX for now. Multiple people looked at the data and some profiles, and the actual reason for the performance regression couldn't really be found. Although it is a 15% change, the absolute numbers are so low that this change still doesn't really matter, as it is way below anything users will perceive.

Since nothing new has shown up so far, the WebCompat GoFaster v2 is officially landed in Firefox 61.

**Comment by miketaylr** [comment]
@denschub can you give an update on where things stand today? thanks!

**Comment by miketaylr** [comment]
status update: for desktop, we expect GREEN status by end of week for 61 beta.
mobile: blocked on getting system addons to work in fennec (link bug here)
site patches: most of them written, just waiting on blockers then will submit PR.

**Comment by denschub** [comment]
Desktop: Approval for 61 arrived, so we're good to go.
Mobile: Status unchanged.
Site patches: Site patches are written, planning on do some careful testing today and pushing to bugzilla tomorrow!

**Comment by denschub** [comment]
Desktop: Unchaned (done)
Mobile: Status unchanged, still blocked.
Site patches: Doing tests, having issues with building, but still should be fine. However, landing won't happen, because PI needs to get involved.

**Comment by miketaylr** [comment]
Let's go ahead and close since the quarter is done. Thanks @denschub!

**Comment by miketaylr** [comment]
Score: 0.5

We managed to ship in Desktop on the 61 train, and Dennis has 4 of 6 site patches written, but PI will need to test them before they can ship. We also ended up blocked for most of the quarter on the Fennec portion of this.
